routers/user/home.go

```diff
@@ -282,26 +271,87 @@ func Issues(ctx *context.Context) {
 		return
 	}
 
-	// Get posters and repository.
-	for i := range issues {
-		issues[i].Repo, err = models.GetRepositoryByID(issues[i].RepoID)
+	var filterIssues []*models.Issue
+	if filterMode == models.FM_ASSIGN {
+		filterIssues = assignedIssues
+	} else if filterMode == models.FM_CREATE {
+		filterIssues = createdIssues
+	} else {
+		filterIssues = yourIssues
+	}
+
+	showRepos    := make([]*models.Repository, 0, len(filterIssues))
+	showReposSet := make(map[int64]bool)
+
+	if repoID > 0 {
+		repo, err := models.GetRepositoryByID(repoID)
+		if err != nil {
+			ctx.Handle(500, "GetRepositoryByID", fmt.Errorf("[#%d]%v", repoID, err))
+			return
+		}
+
+		if err = repo.GetOwner(); err != nil {
+			ctx.Handle(500, "GetOwner", fmt.Errorf("[#%d]%v", repoID, err))
+			return
+		}
+
+		showReposSet[repoID] = true
+		showRepos            = append(showRepos, repo)
+	}
+
+	showIssues   := make([]*models.Issue, 0, len(filterIssues))
+	for _, issue := range filterIssues {
+		// Check if we want to show this issue.
+		if (issue.IsClosed != isShowClosed) ||
+		(repoID > 0 && repoID != issue.RepoID) {
+			continue
+		}
+
+		// Get Repository data.
+		issue.Repo, err = models.GetRepositoryByID(issue.RepoID)
 		if err != nil {
-			ctx.Handle(500, "GetRepositoryByID", fmt.Errorf("[#%d]%v", issues[i].ID, err))
+			ctx.Handle(500, "GetRepositoryByID", fmt.Errorf("[#%d]%v", issue.RepoID, err))
 			return
 		}
 
-		if err = issues[i].Repo.GetOwner(); err != nil {
-			ctx.Handle(500, "GetOwner", fmt.Errorf("[#%d]%v", issues[i].ID, err))
+		// Get Owner data.
+		if err = issue.Repo.GetOwner(); err != nil {
+			ctx.Handle(500, "GetOwner", fmt.Errorf("[#%d]%v", issue.RepoID, err))
 			return
 		}
+
+		// Append to list of issues and repos.
+		showIssues  = append(showIssues, issue)
+		if filterMode == models.FM_YOUR_REPOSITORIES {
+			// Use a map to make sure we don't add the same Repository twice.
+			_, ok := showReposSet[issue.RepoID]
+			if !ok {
+				showReposSet[issue.RepoID] = true
+				// Append to list of shown Repositories.
+				showRepos                  = append(showRepos, issue.Repo)
+			}
+		}
 	}
-	ctx.Data["Issues"] = issues
 
-	ctx.Data["IssueStats"] = issueStats
-	ctx.Data["ViewType"] = viewType
-	ctx.Data["SortType"] = sortType
-	ctx.Data["RepoID"] = repoID
+	var issueStats *models.IssueStats
+	if filterMode == models.FM_YOUR_REPOSITORIES {
+		issueStats = models.GetUserIssueStats(repoID, ctxUser.ID, userRepoIDs, filterMode, isPullList)
+	} else {
+		issueStats = models.GetUserIssueStats(repoID, ctxUser.ID, nil, filterMode, isPullList)
+	}
+	issueStats.YourRepositoriesCount = int64(len(yourIssues))
```

Comment:
Meaning both open and closed issues?

I indeed agree that what is there now is not logical. I'm wondering if it should always be the number of **open** issues / pull requests or the **total** number of issues / pull requests. What do you think?